### PR TITLE
ci: add cargo-audit step

### PR DESCRIPTION
**Pull request by Kazy**
Add a `cargo audit` step to our CI. We probably don't want it to be blocking, to avoid any PR "just" because of a security issue, but this should still be looked at. Not clear how to make it non blocking though, probably from an admin in Github.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ executors:
   docker-rust:
     docker:
       - image: cimg/rust:1.72.0
-    resource_class: small
+    resource_class: medium
```
Comment (jonaro00) [inline .circleci/config.yml:13]: We do not need to bump this

File: .circleci/config.yml [hunk 2]
```diff
@@ -188,6 +188,26 @@ jobs:
...
+    steps:
+      - checkout
+      - run: |
+          echo "$(cargo search cargo-audit | grep '^cargo-audit =' | sha256sum)" > /tmp/cargo-audit-cache-key
+      - restore_cache:
+          name: Restore Cargo audit cache
+          keys: 
+            - cargo-audit-{{ checksum "/tmp/cargo-audit-cache-key" }}
+      - run: |
+          cargo install --force cargo-audit
```
Comment (jonaro00) [inline .circleci/config.yml]: If we download a binary instead, we don't need caching, see for example how sccache is done

**Review by jonaro00 — CHANGES_REQUESTED** [review]
A job failing will not block merging. If red, we just review the output and fix it if needed.

File: .circleci/config.yml (hunk 1 above)
Comment (Kazy) [reply .circleci/config.yml:13]: Oops mistake, didn't want to bump this one.

**Pull request merged**